### PR TITLE
Migrating WT3 landfill zld model

### PR DESCRIPTION
**Pull request by luohezhiming**
## Summary/Motivation:
Add landfill zld model

## Changes proposed in this PR:
- add landfill zld model 
- add corresponding tests

### Legal Acknowledgement

By contributing to this software project, I agree to the following terms and conditions for my contribution:

1. I agree my contributions are submitted under the license terms described in the LICENSE.txt file at the top level of this directory.
2. I represent I am authorized to make the contributions and grant the license. If my employer has rights to intellectual property that includes these contributions, I represent that I have received permission to make contributions and grant the required license on behalf of that employer.

### CODE REVIEW

**Review by kurbansitterley — DISMISSED** [review]
LGTM, but @adam-a-a @bknueven should we be setting `energy_electric_flow_vol_inlet` to zero in .yamls or omit it for units with no energy consumption? Or have we not settled on a convention? I don't think this should hold this up but curious for future development.

File: watertap/data/techno_economic/tests/test_unit_parameter_files.py [hunk 1]
```diff
@@ -55,7 +55,7 @@ def test_unit_parameter_files(tech):
                     "static_mixer",
                     "co2_addition",
                     "sw_onshore_intake",
-                    ]
+                    "landfill_zld"]
```
Comment (bknueven) [inline watertap/data/techno_economic/tests/test_unit_parameter_files.py]:
```suggestion
                    "landfill_zld",
                    ]
```

Leaving the trailing comma helps avoid merge conflicts.

**Comment by bknueven** [comment]
> LGTM, but @adam-a-a @bknueven should we be setting `energy_electric_flow_vol_inlet` to zero in .yamls or omit it for units with no energy consumption? Or have we not settled on a convention? I don't think this should hold this up but curious for future development.

Our general strategy seems to be to omit such things. But I doubt it really matters.

**Review by bknueven — APPROVED** [review]

**Comment by adam-a-a** [comment]
> > LGTM, but @adam-a-a @bknueven should we be setting `energy_electric_flow_vol_inlet` to zero in .yamls or omit it for units with no energy consumption? Or have we not settled on a convention? I don't think this should hold this up but curious for future development.
> 
> Our general strategy seems to be to omit such things. But I doubt it really matters.

In some cases, we've left it in the yaml for cases where we think there should be some electricity consumption but don't have a value for.

**Review by adam-a-a — APPROVED** [review]

**Pull request merged**